lib/Twig/Extensions/Extension/Gettext/Token.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+class Twig_Extensions_Extension_Gettext_Token extends Twig_Token {
+    
+    const COMMENT = 12;
+    
+    public static function typeToString($type, $short = false, $line = -1) {
+        switch ($type) {
+            case self::COMMENT :
+                $name = 'COMMENT';
+                break;
+            default :
+                return parent::typeToString($type, $short, $line);
+        }
+
+        return $short ? $name : 'Twig_Token::'.$name;
```

Comment:
It is patched in right above as `const`. It is necessary to treat comments as their own node type instead of just discarding them, since comments are used to add valuable context in the gettext workflow.